src/ui/pages/Menu/Menu.tsx

```diff
@@ -85,52 +87,66 @@ const Menu = () => {
     );
   };
 
+  const SubMenuChildren = [Settings];
+  const subMenuTitle = ["settings.sections.header"];
+  const subMenuAdditionalButtons = [<></>];
```

Comment:
It would be less error prone if there was a single variable that mapped an enum value to the items as a type.
```
const subMenuItems = new Map([
  [SubMenuKey.SETTINGS, { child: Settings, title: "settings.section.header", additionalButtons: <></>],
]);

...

subMenuItems.get(selectedValue);
```
where `selectedValue` uses a new enum `SubMenuKey` and the objects in the map follow a new type e.g. `SubMenuValue`

Reply:
otherwise we can get a maybe difficult to debug index out of bounds error. The other bug where the indices for each variable doesn't match is probably less likely since it will be tested. But this way is more properly typed